common/membership/ringpop/service_resolver.go

```diff
@@ -347,30 +385,29 @@ func (r *serviceResolver) refreshRingWorker() {
 	}
 }
 
-func (r *serviceResolver) ring() *hashring.HashRing {
-	return r.ringValue.Load().(*hashring.HashRing)
-}
-
-func (r *serviceResolver) getLabelsMap() map[string]string {
-	labels := make(map[string]string)
-	labels[roleKey] = string(r.service)
-	return labels
+func (r *serviceResolver) ring() (*hashring.HashRing, map[string]*hostInfo) {
+	ring := r.ringAndHosts.Load().(ringAndHosts)
```

Comment:
Is it possible for this to be called before the first Store to this atomic?

Reply:
this is used exactly the same as `ringValue` before so if there are bugs they're not new.

to answer the question, though, no, it does a Store in the constructor (new line 117)